Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -151,23 +151,26 @@ namespace
 
             if (inputSource.JointsTrackedThisFrame)
             {
-                auto handJointCollection = Napi::Array::New(env, HAND_JOINT_NAMES.size());
+                auto handJointCollection = jsInputSource.Get("hand").As<Napi::Array>();
+                if (!jsInputSource.Has("hand"))
+                {
+                    handJointCollection = Napi::Array::New(env, HAND_JOINT_NAMES.size());
+
+                    auto jointGetter = [handJointCollection](const Napi::CallbackInfo& info) -> Napi::Value {
+                        return handJointCollection.Get(info[0].As<Napi::String>());
+                    };
+
+                    handJointCollection.Set("get", Napi::Function::New(env, jointGetter, "get"));
+                    handJointCollection.Set("size", static_cast<int>(HAND_JOINT_NAMES.size()));
+
+                    jsInputSource.Set("hand", handJointCollection);
```

Comment:
Lack of experience keeps me from knowing whether capture by reference or value is better here, but I do want the `hand` property to outlive `handJointCollection`, so I think that dependence is fine, right?

Reply:
You definitely don't want to capture by reference (at least not as a C++ reference). If anything, it would be to capture a Napi::Reference. I don't think it's necessary (as you say) hand should outlive handJointCollection.